doc/cprover-manual/smt2-incr.md

```diff
@@ -0,0 +1,40 @@
+[CPROVER Manual TOC](../)
+
+## Incremental SMT backend
+
+CBMC comes with a solver-agnostic incremental SMT backend.
+
+The SMT incremental backend supports the following `C` primitives:
+
+- Bit vector arithmetics
+- Pointers
+- Arrays (support is currently incomplete)
+
+Primitive not supported are:
+
+- Floating point values and arithmetics
+- Structs
+
+Requirement to use the incremental SMT backend is to have a solver which accepts
+SMTlib 2.6 incremental input from the standard input and that supports the
+`QF_AUFBV` (Quantifier Free Array Uninterpreted Functions and Bit Vectors).
+
+To use this functionality it is enough to add the argument
+`--incremental-smt2-solver <cmd>` where `<cmd>` is the command to invoke the SMT
+solver of choice using the interactive mode and accepting input from the
+standard input.
+
+Examples are:
+
+1. `--incremental-smt2-solver 'z3 -smt2 -in'` to use the Z3 solver.
+2. `--incremental-smt2-solver 'cvc5 --lang=smtlib2.6 --incremental'` to use the
+   CVC5 solver.
+
+Note that the solver has to be user-supplied and CBMC does not come packaged
```

Comment:
I believe this information is inaccurate - we do build an SMT2 solver with CBMC (you can find regression tests for that under `regression/smt2_solver` and the binary under `<build>/bin/smt2_solver`) but we are not targeting that as it's very primitive and its current level of SMTlib support is unknown.

I suggest we rephrase that as

> The new incremental SMT backend has been designed to interoperate with external solvers

Reply:
changed